File: 2-add-two-numbers/add-two-numbers.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode addTwoNumbers(ListNode l1, ListNode l2) {
        ListNode node = new ListNode(0);
        ListNode tmp = node;
        int carry = 0;
        while(l1 != null || l2 != null || carry != 0) {
            int value = carry + (l1 != null  ? l1.val : 0) + (l2 != null  ? l2.val : 0);
            carry = value / 10;

            tmp.next = new ListNode(value % 10);
            tmp = tmp.next;
            if(l1 != null) l1 = l1.next;
            if(l2 != null) l2 = l2.next;
        }

        return node.next;
        
    }
}

File: 2-add-two-numbers/add-two-numbers.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def addTwoNumbers(self, l1: Optional[ListNode], l2: Optional[ListNode]) -> Optional[ListNode]:
        dummy = ListNode()
        dummy2 = dummy
        carry = 0
        while (l1 or l2 or carry):
            total = carry + (l1.val if l1 else 0) + (l2.val if l2 else 0)
            newVal = total % 10
            carry = total // 10
            dummy2.next = ListNode(newVal)
            l1 = l1.next if l1 else l1
            l2 = l2.next if l2 else l2

            dummy2 = dummy2.next

        return dummy.next
